teamcity: failed test: TestReplicaGCRace

The following tests appear to have failed on master (testrace): TestReplicaGCRace

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestReplicaGCRace).

[#1553206](https://teamcity.cockroachdb.com/viewLog.html?buildId=1553206):


```
TestReplicaGCRace
...replica r1/3
I191022 14:35:10.306306 157929 storage/replica_destroy.go:164  [s3,r1/3:/{Min-System/NodeL…}] removed 28 (25+3) keys in 8ms [clear=8ms commit=1ms]
I191022 14:35:10.360693 158230 gossip/gossip.go:1531  [n2] node has connected to cluster via gossip
I191022 14:35:10.553660 158242 gossip/gossip.go:1531  [n3] node has connected to cluster via gossip
I191022 14:35:11.329408 159040 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I191022 14:35:11.330326 159039 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I191022 14:35:11.331114 159038 util/stop/stopper.go:542  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I191022 14:35:11.332724 159038 util/stop/stopper.go:542  quiescing; tasks left:
2      mtc send
I191022 14:35:11.335707 159038 util/stop/stopper.go:542  quiescing; tasks left:
1      mtc send
W191022 14:35:11.387659 157858 internal/client/txn.go:518  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I191022 14:35:11.389154 157858 storage/node_liveness.go:846  [liveness-hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
I191022 14:35:11.516225 159037 util/stop/stopper.go:542  quiescing; tasks left:
6      rpc heartbeat
W191022 14:35:11.523053 158610 storage/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W191022 14:35:11.523234 158230 gossip/gossip.go:1517  [n2] no incoming or outgoing connections
W191022 14:35:11.525153 158388 storage/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I191022 14:35:11.527308 159037 util/stop/stopper.go:542  quiescing; tasks left:
4      rpc heartbeat
W191022 14:35:11.524605 158531 storage/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W191022 14:35:11.524193 158430 storage/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W191022 14:35:11.542520 158242 gossip/gossip.go:1517  [n3] no incoming or outgoing connections
I191022 14:35:11.542818 159037 util/stop/stopper.go:542  quiescing; tasks left:
2      rpc heartbeat
panic: only messages with coalesced heartbeats or heartbeat responses may be sent to range ID 0 [recovered]

goroutine 157929 [running]:
testing.tRunner.func1(0xc001fe8300)
panic(0x5440d40, 0x68c59e0)
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
panic(0x5440d40, 0x68c59e0)
github.com/cockroachdb/cockroach/pkg/storage.(*RaftTransport).SendAsync(0xc0018c2240, 0xc0009f6000, 0x0, 0x0)
github.com/cockroachdb/cockroach/pkg/storage_test.TestReplicaGCRace(0xc001fe8300)
testing.tRunner(0xc001fe8300, 0x5b40038)
created by testing.(*T).Run

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0001a2500, 0x59dc932, 0x11, 0x5b40038, 0x1)
testing.runTests.func1(0xc0001a2500)
testing.tRunner(0xc0001a2500, 0xc000513b20)
testing.runTests(0xc0002de3e0, 0x9505300, 0x24f, 0x24f, 0x0)
testing.(*M).Run(0xc0003e4800, 0x0)
github.com/cockroachdb/cockroach/pkg/storage_test.TestMain(0xc0003e4800)
main.main()

goroutine 6 [syscall, 9 minutes]:
os/signal.signal_recv(0xc0a151)
os/signal.loop()
created by os/signal.init.0

goroutine 26 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
created by github.com/cockroachdb/cockroach/pkg/util/log.init.7

goroutine 27 [chan receive, 9 minutes]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
created by github.com/cockroachdb/cockroach/pkg/util/log.init.7

goroutine 7 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc0002a8700)
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -2879,7 +2879,11 @@ func TestReplicaGCRace(t *testing.T) {
 	// Send the heartbeat. Boom. See #11591.
 	// We have to send this multiple times to protect against
 	// dropped messages (see #18355).
-	if sent := fromTransport.SendAsync(&hbReq, rpc.DefaultClass); !sent {
+	sendHeartbeat := func() (sent bool) {
+		r := hbReq
+		return fromTransport.SendAsync(&r, rpc.DefaultClass)
+	}
+	if sent := sendHeartbeat(); !sent {
 		t.Fatal("failed to send heartbeat")
 	}
 	heartbeatsSent := 1
@@ -2899,7 +2903,7 @@ func TestReplicaGCRace(t *testing.T) {
 			t.Fatal("did not get expected error")
 		}
 		heartbeatsSent++
-		if sent := fromTransport.SendAsync(&hbReq, rpc.DefaultClass); !sent {
+		if sent := sendHeartbeat(); !sent {
 			t.Fatal("failed to send heartbeat")
 		}
 	}

